### PR TITLE
Static test cases definition + supporting multi-model cases in MemLeaks tests

### PR DESCRIPTION
**Pull request by alexander-shchepetov**

### CODE REVIEW

File: tests/stress_tests/.automation/memleaks_tests/nightly_configs/desktop_test_config.xml [hunk 1]
```diff
@@ -1,21 +1,14 @@
...
-        <model name="alexnet" precision="FP32" source="omz" />
-        <model name="mobilenet-ssd" precision="FP32" source="omz" />
-        <model name="mtcnn-r" precision="FP32" source="omz" />
-    </models>
-</attributes>
\ No newline at end of file
+<cases>
+    <device iterations="1000" name="CPU" processes="1" threads="1">
+        
+        <model framework="caffe" full_path="/home/my/Desktop/repos/openvino-as-fork/tests/stress_tests/_omz_out/irs/public/alexnet/FP32/alexnet.xml" name="alexnet" path="public/alexnet/FP32/alexnet.xml" precision="FP32" source="omz" subdirectory="public/alexnet" />
```
Comment (ishariko) [inline tests/stress_tests/.automation/memleaks_tests/nightly_configs/desktop_test_config.xml:4]: Need to updated path

Comment (alexander-shchepetov) [reply tests/stress_tests/.automation/memleaks_tests/nightly_configs/desktop_test_config.xml:4]: @ishariko, I accidentally did this :)

File: tests/stress_tests/.automation/memleaks_tests/nightly_configs/desktop_test_config.xml [hunk 2]
```diff
@@ -1,21 +1,14 @@
...
+    </device>
+    <device iterations="1000" name="GPU" processes="1" threads="1">
+        
+        <model framework="caffe" full_path="/home/my/Desktop/repos/openvino-as-fork/tests/stress_tests/_omz_out/irs/public/alexnet/FP32/alexnet.xml" name="alexnet" path="public/alexnet/FP32/alexnet.xml" precision="FP32" source="omz" subdirectory="public/alexnet" />
+        <model framework="caffe" full_path="/home/my/Desktop/repos/openvino-as-fork/tests/stress_tests/_omz_out/irs/public/mobilenet-ssd/FP32/mobilenet-ssd.xml" name="mobilenet-ssd" path="public/mobilenet-ssd/FP32/mobilenet-ssd.xml" precision="FP32" source="omz" subdirectory="public/mobilenet-ssd" />
+        <model framework="caffe" full_path="/home/my/Desktop/repos/openvino-as-fork/tests/stress_tests/_omz_out/irs/public/mtcnn/mtcnn-r/FP32/mtcnn-r.xml" name="mtcnn-r" path="public/mtcnn/mtcnn-r/FP32/mtcnn-r.xml" precision="FP32" source="omz" subdirectory="public/mtcnn/mtcnn-r" />
+    </device>
+</cases>
```
Comment (ishariko) [inline tests/stress_tests/.automation/memleaks_tests/nightly_configs/desktop_test_config.xml:14]: please, add empty line in the end

File: tests/stress_tests/scripts/get_testdata.py [hunk 3]
```diff
@@ -94,6 +107,9 @@ def main():
     parser.add_argument('--test_conf', required=True, type=Path,
                         help='Path to a test config .xml file containing models '
                              'which will be downloaded and converted to IRs via OMZ.')
+    parser.add_argument('--test_framework', required=False, default=None,
```
Comment (asomsiko) [inline tests/stress_tests/scripts/get_testdata.py:110]: Why framework selection is needed? Can we autodetect it?

**Review by asomsiko — CHANGES_REQUESTED** [review]

File: tests/stress_tests/.automation/memleaks_tests/nightly_configs/desktop_test_config.xml (hunk 2 above)
Comment (alexander-shchepetov) [reply tests/stress_tests/.automation/memleaks_tests/nightly_configs/desktop_test_config.xml]: Done

File: tests/stress_tests/common/tests_utils.h [hunk 4]
```diff
@@ -22,16 +22,29 @@ enum TestStatus
...
+        }
+        return _item;
+    }
+};
+
+class TestCase: public TestCaseBase {
+public:
+    std::string model;
 
     TestCase(int _numprocesses, int _numthreads, int _numiters, std::string _device, const std::string& _model, const std::string& _model_name, const std::string& _precision) {
```
[the remark this answers is not in the document]
Comment (alexander-shchepetov) [reply tests/stress_tests/common/tests_utils.h:50]: Done

File: tests/stress_tests/scripts/get_testdata.py (hunk 3 above)
Comment (alexander-shchepetov) [reply tests/stress_tests/scripts/get_testdata.py]: Yes, we can, I done it

File: tests/stress_tests/common/tests_utils.h [hunk 5]
```diff
@@ -22,16 +22,29 @@ enum TestStatus
...
     int numiters;
-    std::string device;
-    std::string model_name;
-    std::string model;
     std::string precision;
     std::string test_case_name;
+    std::string model_name;
+    std::string device;
+protected:
+    std::string update_item_for_name(const std::string &item) {
```
[the remark this answers is not in the document]
Comment (alexander-shchepetov) [reply tests/stress_tests/common/tests_utils.h:36]: Done

File: tests/stress_tests/common/tests_utils.h [hunk 6]
```diff
@@ -40,15 +53,23 @@ class TestCase {
...
+
+    MemLeaksTestCase(int _numprocesses, int _numthreads, int _numiters, std::string _device, std::vector<std::map<std::string, std::string>> _models) {
+        numprocesses = _numprocesses, numthreads = _numthreads, numiters = _numiters, device = _device, models = _models;
+        test_case_name =
+                "Numprocesses_" + std::to_string(numprocesses) + "_Numthreads_" + std::to_string(numthreads) +
+                "_Numiters_" + std::to_string(numiters) + "_Device_" + update_item_for_name(device);
+        for (int i = 0; i < models.size(); i++){
+            test_case_name +=
+                    "_Model" + std::to_string(i) + "_" + update_item_for_name(models[i]["name"]) +
+                    "_Precision_" +  update_item_for_name(models[i]["precision"]);
```
[the remark this answers is not in the document]
Comment (alexander-shchepetov) [reply tests/stress_tests/common/tests_utils.h]: I just shortened test cases names

File: tests/stress_tests/common/tests_utils.cpp [hunk 7]
```diff
@@ -75,13 +76,50 @@ std::vector<TestCase> generateTestsParams(std::initializer_list<std::string> fie
                 for (auto &device : devices)
                     for (int i = 0; i < models.size(); i++)
                         tests_cases.push_back(TestCase(numprocesses, numthreads, numiters, device, models[i], models_names[i], precisions[i]));
+                    return tests_cases;
+}
+
+std::vector<MemLeaksTestCase> generateTestsParamsMemLeaks() {
```
[the remark this answers is not in the document]
Comment (alexander-shchepetov) [reply tests/stress_tests/common/tests_utils.cpp:81]: Done

**Review by just-sparta — APPROVED** [review]

File: tests/stress_tests/scripts/get_testdata.py (hunk 3 above)
Comment (asomsiko) [reply tests/stress_tests/scripts/get_testdata.py]: I'd expect you to remove that option then.

**Comment by alexander-shchepetov** [comment]
https://openvino-ci.intel.com/job/private-ci/job/ie/job/stress-tests-linux/1810/

**Review by asomsiko — APPROVED** [review]

**Review by ishariko — APPROVED** [review]

**Pull request merged**